src/AppBundle/Service/UserRegistrationService.php

```diff
@@ -25,39 +25,49 @@ public function __construct($em)
 
     /**
      * CASREC checks
+     * - throw error 403 if user is a co-deputy attempting to self-register
      * - throw error 421 if user and client not found
      * - throw error 422 if user email is already found
      * - throw error 424 if user and client are found but the postcode doesn't match
      * - throw error 425 if client is already used
      * (see <root>/README.md for more info. Keep the readme file updated with this logic).
      *
+     * @param SelfRegisterData $selfRegisterData
      * @return User
      */
     public function selfRegisterUser(SelfRegisterData $selfRegisterData)
     {
-        $user = new User();
-        $client = new Client();
+        $existingClient = $this->em->getRepository('AppBundle\Entity\Client')->findOneBy(['caseNumber' => CasRec::normaliseCaseNumber($selfRegisterData->getCaseNumber())]);
 
-        $user->recreateRegistrationToken();
-        $this->populateUser($user, $selfRegisterData);
-        $this->populateClient($client, $selfRegisterData);
+        // ward off non-fee-paying codeps trying to self-register
+        if ($this->isMultiDeputyCase($selfRegisterData->getCaseNumber()) && $existingClient instanceof Client) {
+            // if client exists with case number, the first codep already registered.
+            throw new \RuntimeException("Co-deputy cannot self register.", 403);
+        }
 
-        // Check the user is unique
-        if (!$this->userIsUnique($user)) {
-            throw new \RuntimeException("User with email {$user->getEmail()} already exists.", 422);
+        // Check the user doesn't already exist
+        $existingUser = $this->em->getRepository('AppBundle\Entity\User')->findOneBy(['email' => $selfRegisterData->getEmail()]);
+        if ($existingUser) {
+            throw new \RuntimeException("User with email {$existingUser->getEmail()} already exists.", 422);
         }
 
         // Check the client is unique
-        if (!$this->clientIsUnique($client)) {
```

Comment:
Should remove the comment too if unique check is removed

Reply:
no - the check still takes place at that point - just refactored